source/acceptance_test.go

```diff
@@ -0,0 +1,234 @@
+	// suppress false positive goroutine leaks.
+	GoleakOptions []goleak.Option
+
+	// Skip is a slice of regular expressions used to identify tests that should
+	// be skipped. The full test name will be matched against all regular
+	// expressions and the test will be skipped if a match is found.
+	Skip []string
+}
+
+func (d AcceptanceTestDriver) DestinationConfig(*testing.T) map[string]string {
+	return map[string]string{}
+}
+func (d AcceptanceTestDriver) Connector() sdk.Connector {
+	return d.Config.Connector
+}
+
+func (d AcceptanceTestDriver) SourceConfig(*testing.T) map[string]string {
+	return d.Config.SourceConfig
+}
+
+func (d AcceptanceTestDriver) BeforeTest(t *testing.T) {
+	// before test check if the test should be skipped
+	d.Skip(t)
+
+	if d.Config.BeforeTest != nil {
+		d.Config.BeforeTest(t)
+	}
+}
+
+func (d AcceptanceTestDriver) AfterTest(t *testing.T) {
+	if d.Config.AfterTest != nil {
+		d.Config.AfterTest(t)
+	}
+}
+
+func (d AcceptanceTestDriver) Skip(t *testing.T) {
+	var skipRegexs []*regexp.Regexp
+	for _, skipRegex := range d.Config.Skip {
+		r := regexp.MustCompile(skipRegex)
+		skipRegexs = append(skipRegexs, r)
+	}
+
+	for _, skipRegex := range skipRegexs {
+		if skipRegex.MatchString(t.Name()) {
+			t.Skipf("caller requested to skip tests that match the regex %q", skipRegex.String())
+		}
+	}
+}
+
+func (d AcceptanceTestDriver) GoleakOptions(_ *testing.T) []goleak.Option {
+	return d.Config.GoleakOptions
+}
+
+// WriteToSource writes data for source to pull data from
+func (d AcceptanceTestDriver) WriteToSource(t *testing.T, records []sdk.Record) []sdk.Record {
+	var err error
+	is := is.New(t)
+	client, err := getClient()
+	is.NoErr(err)
+	leads, err := addLeads(client, len(records))
+	is.NoErr(err)
+	var emailIDs = make([]string, 0)
+	for _, v := range leads {
+		emailIDs = append(emailIDs, v["email"].(string))
+	}
+	records, err = writeRecords(client, emailIDs)
+	if err != nil {
+		t.Error(err)
+	}
+
+	return records
+}
+
+func (d AcceptanceTestDriver) ReadFromDestination(*testing.T, []sdk.Record) []sdk.Record {
+	return []sdk.Record{}
+}
+
+func writeRecords(client Client, emailIDs []string) ([]sdk.Record, error) {
+	data, err := client.filterLeads("email", emailIDs)
+	if err != nil {
+		return nil, err
+	}
+	var record []map[string]interface{}
+	err = json.Unmarshal(*data, &record)
+	if err != nil {
+		return nil, err
+	}
+	var records = make([]sdk.Record, 0)
+	for _, v := range record {
+		id := int(v["id"].(float64))
+		data, err = client.getLeadByID(id, Fields)
+		if err != nil {
+			return nil, err
+		}
+		var rec []map[string]interface{}
+		err = json.Unmarshal(*data, &rec)
+		if err != nil {
+			return nil, err
+		}
+		rec[0]["id"] = fmt.Sprint(id)
+		r, err := prepareRecord(rec[0])
+		if err != nil {
+			return nil, err
+		}
+		records = append(records, r)
+	}
+	return records, err
+}
+
+func prepareRecord(data map[string]interface{}) (sdk.Record, error) {
```

Comment:
Unless I'm mistaken, this is almost an exact copy of CDCIterator.prepareRecord?

Reply:
not exactly, in this method we are not handling delete records